waku/protocol/v2/waku_store.nim

```diff
@@ -90,3 +96,17 @@ proc subscription*(proto: WakuStore): MessageNotificationSubscription =
     proto.messages.add(msg)
 
   MessageNotificationSubscription.init(@[], handle)
+
+proc query*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc) {.async, gcsafe.} =
+  let conn = await w.switch.dial(w.peerInfo.peerId, w.peerInfo.addrs, WakuStoreCodec)
```

Comment:
This seems to imply we can just dial any peer. I suppose it is fine for now, but it'd be nice with a data structure that captures some more info about the peer. E.g. allows one to answer Qs like:

- is it our default store peer?
- which topics does it store on?
- online time?
- latency/preference, etc

At least put as TODO if we want to do it after. This is important for newcomer joining as well, so I'd probably create a wrapper data structure `seq[HistoryPeer]` (?) with one field or so for now (`.peerInfo`). Something like this

Reply:
e.g. a bit like Subscriber object in filter pr